app/src/main/res/layout/activity_announcements.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.coordinatorlayout.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <include layout="@layout/toolbar_basic" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        app:layout_behavior="@string/appbar_scrolling_view_behavior">
```

Comment:
It should be possible to do this layout without a `ConstraintLayout`. Try putting the `app:layout_behavior="@string/appbar_scrolling_view_behavior"` on the `SwipeRefreshLayout` and center the `Progressbar` and the `BackgroundMessageView` inside the `CoordinatorLayout`.